androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/TrackUtils.kt

```diff
@@ -0,0 +1,89 @@
+package com.profusion.androidenhancedvideoplayer.utils
+
+import androidx.annotation.OptIn
+import androidx.media3.common.C
+import androidx.media3.common.Format
+import androidx.media3.common.TrackGroup
+import androidx.media3.common.TrackSelectionOverride
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.trackselection.DefaultTrackSelector
+import androidx.media3.exoplayer.trackselection.TrackSelector
+
+@OptIn(UnstableApi::class)
+fun DefaultTrackSelector.getAutoTrack(label: String? = null): TrackQualityItem {
+    return TrackQualityItem(
+        builder = buildUponParameters()
+            .clearOverridesOfType(C.TRACK_TYPE_VIDEO)
+            .setAllowMultipleAdaptiveSelections(true),
+        autoLabel = label
+    )
+}
+
+fun generateTrackQualityOptions(trackSelector: TrackSelector? = null): List<TrackQualityItem> {
+    return if (trackSelector is DefaultTrackSelector) {
+        trackSelector.generateTrackQualityOptions()
+    } else {
+        listOf()
+    }
+}
+
+@OptIn(UnstableApi::class)
+fun DefaultTrackSelector.generateTrackQualityOptions():
+    List<TrackQualityItem> {
+    val qualityOptions = ArrayList<TrackQualityItem>()
+
+    val videoQualityOptions = getAvailableVideoTracks().map { (trackIndex, trackGroup) ->
+        TrackQualityItem(
+            format = trackGroup.getFormat(trackIndex),
+            builder = buildUponParameters()
+                .clearOverridesOfType(C.TRACK_TYPE_VIDEO)
+                .addOverride(
+                    TrackSelectionOverride(trackGroup, listOf(trackIndex))
+                )
```

Comment:
```suggestion
            builder = buildUponParameters()
                .clearOverridesOfType(C.TRACK_TYPE_VIDEO)
                .addOverride(
                    TrackSelectionOverride(trackGroup, listOf(trackIndex))
                )
                .build()
```

Reply:
Yeah. It is better to pass the builded object instead of a builder.